src/crimson/osd/shard_services.cc

```diff
@@ -188,21 +188,22 @@ seastar::future<> ShardServices::send_pg_temp()
   if (pg_temp_wanted.empty())
     return seastar::now();
   logger().debug("{}: {}", __func__, pg_temp_wanted);
-  boost::intrusive_ptr<MOSDPGTemp> ms[2] = {nullptr, nullptr};
+  MURef<MOSDPGTemp> ms[2] = {crimson::net::make_null_message<MOSDPGTemp>(), 
```

Comment:
Huh, haven't seen `make_null_message` in any previous commit. Do we already have it?

Anyway, why not just `nullptr`?

Reply:
just nullptr doesn't work as MURef is not using the default deleter

Reply:
Hmm, according to the doc for `std::unique_ptr<T>::unique_ptr(std::nullptr_t)`:

> Constructs a `std::unique_ptr` that owns nothing. Value-initializes the stored pointer and the stored deleter. **Requires that `Deleter `is DefaultConstructible** and that construction does not throw an exception. These overloads participate in overload resolution only if `std::is_default_constructible<Deleter>`::value is true **and `Deleter` is not a pointer type**.

In short: it might be worth consideration to switch the `unique_ptr_release` into a callable class (_functor_). The pros is also getting rid of the deleter-as-pointer's overhead.